android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/AuthenticationManager.java

```diff
@@ -211,34 +342,73 @@ public Completable resendEmailVerification(@NonNull final String email) {
+                        return Single.just(((ConsentRequiredException) t).getSession());
+                    }
+                    return Single.error(t);
+                })
+                .doOnSuccess(session -> {
+                    accountDAO.setEmail(email);
+
+                    // if we signed in with a password, save it
+                    String password = signIn.getPassword();
+                    if (!Strings.isNullOrEmpty(password)) {
+                        accountDAO.setPassword(password);
+                    }
+
+                    accountDAO.setUserSessionInfo(session);
+
+                    authStateHolderAtomicReference.set(
+                            createAuthStateFromStoredCredentials()
                     );
-                    accountDAO.setUserSessionInfo(userSessionInfo);
+
                     accountDAO.setStudyParticipant(
                             new StudyParticipant()
                                     .email(signIn.getEmail()));
-                    for (AuthenticationEventListener listener : listeners) {
-                        listener.onSignedIn(email);
-                    }
-                }).doOnError(throwable -> {
-                    // a 412 is a successful signin
-                    if (throwable instanceof ConsentRequiredException) {
-                        forConsentedUsersApiAtomicReference.set(
-                                apiClientProvider.getClient(ForConsentedUsersApi.class, signIn)
-                        );
-                        accountDAO.setUserSessionInfo(
-                                ((ConsentRequiredException) throwable).getSession());
-                        accountDAO.setStudyParticipant(
-                                new StudyParticipant()
-                                        .email(signIn.getEmail()));
-                    } else {
-                        accountDAO.setSignIn(null);
+
+                    if (!session.getConsented()) {
+                        // look for a missing required consent which we have locally
+                        for (Map.Entry<String, ConsentStatus> consentStatusEntry : session
+                                .getConsentStatuses().entrySet()) {
+                            String subpopulationGuid = consentStatusEntry.getKey();
+                            ConsentStatus consentStatus = consentStatusEntry.getValue();
+
+                            // required consent missing on Bridge and present locally
+                            if (consentStatus.getRequired()
+                                    && !consentStatus.getConsented()
+                                    && isConsentedInLocal(subpopulationGuid)) {
+
+                                // upload local consents, ignoring errors
+                                uploadLocalConsents()
+                                        .onErrorResumeNext(Observable.just(session))
```

Comment:
Do these errors get caught and sent to this part?...

.doOnError(t -> {
 +                    accountDAO.setSignIn(null);
 +                    accountDAO.setPassword(null);
 +                    accountDAO.setUserSessionInfo(null);
 +
                  });

Reply:
If you're asking about uploadLocalConsents(), no. The onErrorResumeNext will just return the same session

Reply:
For all intents and purposes, wouldn't that be considered a fail on the original network request "sign in"?